modules/sdk-coin-hbar/test/unit/hbar.ts

```diff
@@ -939,4 +939,33 @@ describe('Hedera Hashgraph:', function () {
       );
     });
   });
+
+  describe('Key Derivation', function () {
+    it('seedDerivationPath should be correct', async function () {
+      const expectedDerivationPath = 'm/999999/25725073/5121434';
+      const expectedPub = '302a300506032b657003210044a0c69e1b13ce87410402c96c62bfdccf315ea150d1537b6a43d47acbef60fc';
+      const expectedPrv =
+        '302e020100300506032b657004220420245072d243a8e4232a33ecf9578ecb7ed926926edcb7993318f48f45a7cf7e02';
+
+      // const rootKeyPair = basecoin.createRootKeys()
+
+      const rootKeyPair = {
+        prv: 'rprvd011c015c8ca989b36a5ee7b73661dc854c50a2cb8e91db73f36d76b97cf7e42:9138336fdba9096b6343b9c8b444baf084528725d58a78531607b8534613e41d:8623583b56ebbe8bbd80e6599682e2b1242cabe97cbe013575db04039c028687',
+        pub: 'rpubb765e558aff0fdab8ed42e1745a0fb2ebc2d13512701fad02b147b5c01503eba:9138336fdba9096b6343b9c8b444baf084528725d58a78531607b8534613e41d',
+      };
+      const seed = 'cold derivation seed';
+
+      const derivedPub = await basecoin.deriveKeyWithSeed({ key: rootKeyPair.pub, seed });
+      assert.ok(derivedPub.derivationPath === expectedDerivationPath);
+      assert.ok(derivedPub.key === expectedPub);
+
+      const derivedPrv = await basecoin.deriveKeyWithSeed({ key: rootKeyPair.prv, seed });
+      assert.ok(derivedPrv.derivationPath === expectedDerivationPath);
+      derivedPrv.key.should.equal(expectedPrv);
+
+      const derivedKeychain = new KeyPair({ prv: derivedPrv.key }).getKeys();
+      derivedKeychain.pub.should.equal(expectedPub);
```

Comment:
Fails here, why the derivedKeychain from prv doesnt generate the same pub as the derivation from a pub.

Reply:
@islamaminBitGo  This derivedPrv is the "hbar" formatted prv, appending the pub to the doesnt work.

Reply:
well the fix would be applied to the hbar.ts as im outlining below not here